File: common/src/main/java/com/example/common/validator/StringValidator.java
package com.example.common.validator;

public class StringValidator implements Validator {

    private final String string;
    private boolean result;

    public StringValidator(String string) {
        this.result = true;
        this.string = string;
    }

    public StringValidator notNull() {
        this.result &= this.string != null;
        return this;
    }

    public StringValidator sizeEqualTo(int size) {
        this.result &= this.string != null && this.string.length() == size;
        return this;
    }

    public StringValidator sizeLessOrEqualTo(int size) {
        this.result &= this.string != null && this.string.length() <= size;
        return this;
    }

    public StringValidator sizeGreaterOrEqualTo(int size) {
        this.result &= this.string != null && this.string.length() >= size;
        return this;
    }

    @Override
    public boolean isValid() {
        return this.result;
    }
}

File: services/common/src/main/kotlin/com/example/common/validator/StringValidator.kt
package com.example.common.validator

class StringValidator(private val string: String): Validator {

    private var result = true

    fun notEmpty(): StringValidator {
        result = result and string.trim().isNotEmpty()
        return this
    }

    fun sizeEqualTo(size: Int): StringValidator {
        result = result and (string.length == size)
        return this
    }

    fun sizeLessOrEqualTo(size: Int): StringValidator {
        result = result and (string.length <= size)
        return this
    }

    fun sizeGreaterOrEqualTo(size: Int): StringValidator {
        result = result and (string.length >= size)
        return this
    }

    fun regex(pattern: String): StringValidator {
        result = result and string.matches(Regex.fromLiteral(pattern))
        return this
    }

    override fun isValid(): Boolean {
        return result
    }
}